### PR TITLE
Fix warnings about non-tuple indices in test suite.

### PR DESCRIPTION
**Pull request by insertinterestingnamehere**
Currently a work in progress.

Basically just trying to make the CI tests that use the latest numpy work again.

### CODE REVIEW

**Comment by insertinterestingnamehere** [comment]
The warnings being treated as errors are more widespread than I thought. They are definitely coming from https://github.com/numpy/numpy/pull/9686 though. I'll see if I can revisit this over the next few days unless someone beats me to it. Here's a hopefully exhaustive list of where they are coming from:

numpy/lib/index_tricks.py:204
scipy/interpolate/interpolate.py:2073
scipy/interpolate/interpolate.py:2514
scipy/interpolate/tests/test_interpolate.py:2549
scipy/io/matlab/miobase.py:414
numpy/matrixlib/defmatrix.py:179
scipy/ndimage/morphology.py:114
scipy/signal/_arraytools.py:45
scipy/signal/signaltools.py:491
scipy/signal/signaltools.py:3463
scipy/signal/signaltools.py:2383
scipy/signal/signaltools.py:1593
scipy/signal/spectral.py:278
scipy/signal/signaltools.py:1344
scipy/stats/stats.py:1715
scipy/signal/signaltools.py:2223
scipy/signal/signaltools.py:1341
scipy/spatial/tests/test_distance.py:311
scipy/stats/_binned_statistic.py:607
scipy/stats/stats.py:2710
scipy/stats/stats.py:2833

Note: some of these will have to be fixed upstream.

File: scipy/fftpack/basic.py [hunk 1]
```diff
@@ -147,14 +147,17 @@ def _asfarray(x):
...
     s = list(x.shape)
+    if axis >= len(s) or axis < - len(s):
+        raise IndexError("Input array does not have specified axis.")
+    axis = axis % len(s)
     if s[axis] > n:
-        index = [slice(None)]*len(s)
-        index[axis] = slice(0,n)
+        index = tuple(slice(None) if i != axis else slice(0, n)
+                      for i in range(len(s)))
         x = x[index]
```
Comment (eric-wieser) [inline scipy/fftpack/basic.py]: I'd expect it to be faster to just do `x[tuple(index)]` here - numpy < 1.13 did that internally anyway

**Comment by eric-wieser** [comment]
Upstream fix is at numpy/numpy#11182

**Comment by eric-wieser** [comment]
`numpy/matrixlib/defmatrix.py:179` is correct already - the error is coming from the caller.

**Comment by pv** [comment]
Please add (maybe in a separate PR)

always:Using a non-tuple sequence for multidimensional indexing.*:FutureWarning

to `filterwarnings` in pytest.ini to make the warning non-fatal. Pytest
will still list the occurrences, but they won't fail the test suite.

**Comment by pv** [comment]
... and maybe separate line for PendingDeprecationWarning or omit the
warning class completely, in case it gets changed in numpy.

**Comment by larsoner** [comment]
@insertinterestingnamehere do you have time to take this back up? If not, I'm happy to take over. It would be good to squash the `fftpack` warnings.

**Comment by insertinterestingnamehere** [comment]
@larsoner go for it. I'm only intermittently available at the moment, so that would be great.

**Comment by larsoner** [comment]
Closing for #8944

**Pull request closed without merging**